src/atoms/Heading/HeadingWithTag.js

```diff
@@ -3,16 +3,14 @@ import PropTypes from 'prop-types';
 
 const HeadingWithTag = (props) => {
   const level = props.nodeData.attrs.level;
-  const content = props.nodeData.content[0].text
-    .toLowerCase()
-    .replace(/[^a-z0-9+]+/gi, '');
   const HeadingTag = `h${level}`;
+  const txt = props.nodeData.content.find((content) => content.type === 'text')
```

Comment:
There's a new linkText attr that can be used in a similar manner to the anchor attr below.

Reply:
Will give that a go  thanks

Reply:
So this is not the place to use that though  this outputs the actual `<h3 id="something">Something</h3>`not the links in the TOC

Reply:
I do use linkText here https://github.com/APMG/amat-react/blob/0fd6ab6fce8f187d3edc606eb9cc851aa2995553/src/atoms/ApmTableOfContents/ApmTableOfContents.js#L37

Reply:
whoops. My bad for cruising too fast.

Reply:
Looking at it again is it worth a refactor to only grab the txt if you don't have the anchor or does that get used somewhere else I'm missing?

Reply:
Not sure if we are on the same page here.  It seems to me I always have to get the text.  Maybe we aren't talking about the same thing.  I always have to output the text   `<HeadingTag id={anchor}>{txt}</HeadingTag>;`

I suspect we are both talking about something different